app/assets/javascripts/initializers/initScrolling.js.erb

```diff
@@ -179,40 +179,46 @@ function fetchNextPodcastPage(el) {
   fetchNext(el, "/api/podcast_episodes", insertArticles)
 }
 
-function paginate(tag, requiresApproval) {
-  var tagFilters = [];
-  if (tag && tag.length > 0) {
-    tagFilters.push(tag);
-  }
+function paginate(tag, params, requiresApproval) {
+  const searchHash = { ...{ per_page: 15, page: nextPage }, ...JSON.parse(params) };
 
-  var searchObj = {
-      hitsPerPage: 15,
-      page: nextPage,
-      attributesToHighlight: [],
-      tagFilters,
-      filters: (requiresApproval === 'true') ? 'approved:true' : ''
+  if (tag && tag.length > 0) {
+    searchHash["tag_names"] = searchHash["tag_names"] || [];
+    searchHash["tag_names"].push(tag);
   }
+  searchHash["approved"] = (requiresApproval === 'true') ? 'true' : '';
 
   var homeEl = document.getElementById("index-container");
-  var paginationArticlesIndex;
   if (homeEl.dataset.feed === "base-feed") {
-    paginationArticlesIndex = client.initIndex('ordered_articles_<%= Rails.env %>');
+    searchHash["class_name"] = "Article";
   } else if (homeEl.dataset.feed === "latest") {
-    paginationArticlesIndex = client.initIndex('ordered_articles_by_published_at_<%= Rails.env %>');
+    searchHash["class_name"] = "Article";
+    searchHash["sort_by"] = "published_at";
   } else {
-    var preamble = (requiresApproval === 'true') ? ' AND ' : ''
-    searchObj.filters += ( preamble +'published_at_int > ' + homeEl.dataset.articlesSince);
-    paginationArticlesIndex = client.initIndex('ordered_articles_by_positive_reactions_count_<%= Rails.env %>');
+    searchHash["class_name"] = "Article";
+    searchHash["published_at[gte]"] = homeEl.dataset.articlesSince;
+    searchHash["sort_by"] = "positive_reactions_count";
   }
 
-  paginationArticlesIndex.search("*", searchObj)
-  .then(function searchDone(content) {
+  const searchParams = new URLSearchParams(searchHash).toString();
+
+  fetch(`/search/feed_content?${searchParams}`, {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'X-CSRF-Token': window.csrfToken,
+      'Content-Type': 'application/json',
+    },
+    credentials: 'same-origin',
+  })
+    .then(response => response.json())
```

Comment:
Same here 👀 

Can we use the [`fetchSearch` helper function](https://github.com/atsmith813/dev.to/blob/9a0a6fbe8f3b45df600ba3a1571ab8164fd1bc41/app/javascript/src/utils/search.js#L90-L102) here?

```javascript
const responsePromise = fetchSearch('feed_content', searchHash)

responsePromise.then((content) => {
  // continuing on...
}
```